core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,6 +64,14 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // TODO(davies): configurable
+  if (rdds.length > 10) {
+    // prepare the partitions in parallel (will be cached)
+    val parArray = rdds.toParArray
+    parArray.tasksupport = new ForkJoinTaskSupport(new ForkJoinPool(10))
+    parArray.foreach(_.partitions)
+  }
+
```

Comment:
I actually prefer how it is because if we construct a single collection, we will need to preserve the global order of partitions across RDDs. But since parallel collection doesn't guarantee any order, this will be hard to implement. However, if we cache partitions in RDDs where they belong to, we don't have to worry about the order.

But I made this logic more explicit by defining a ``lazy val`` of ``Unit`` type that gets called by ``getPartitions``. Hopefully, this addresses your concerns.